### PR TITLE
Refactor condition summary

### LINKED ISSUE #2184: See all conditions in eCR Summary | eCR Viewer
## User Story

As an epi or case investigator, I want to see condition summaries for all reportable conditions in an eCR, so that I can understand the full context of why the eCR was triggered and sent to me

## Acceptance Criteria

- [x] For any number of reportable conditions, and for both integrated and non-integrated viewer, I now see accordion design for the data under `Condition Summary`

The accordion design should function as follows:
- [x] The reportable condition should be displayed as the text within the accordion header
- [x] When there is only 1 reportable condition, the accordion is open by default
- [x] When there is more than 1 reportable condition AND we get a SNOMED condition code from NBS, the matching reportable condition is at the top of the list of conditions and expanded by default
- [x] When there is more than 1 reportable condition BUT we don't get a SNOMED condition code from NBS, the conditions are organized in a list alphabetically and all are collapsed by default
- [x] When you click on a collapsed accordion, the previously open accordion closes

## Scenarios

Scenario 1:
- I open an eCR with just 1 reportable condition in the Viewer
- I see an expanded accordion under `Condition Summary` with relevant info about that condition

Scenario 2:
- I open an eCR with n>1 reportable conditions in the Viewer AND pass in a SNOMED condition code that corresponds to one of the reportable conditions (to replicate NBS giving us a code)
- I see n accordion headers
- The accordion header corresponding to the reportable condition code I passed in is at the top of the list of headers and expanded by default
- I click into a collapsed accordion header
- The previously open accordion collapses and the accordion header I just clicked on expands

Scenario 3:
- I open an eCR with n>1 reportable conditions in the Viewer without passing in a condition code
- I see n accordion headers
- All accordion headers are collapsed
- I click into a collapsed accordion header and it expands
- I click into another collapsed accordion header
- The previously open accordion collapses and the accordion header I just clicked on expands

## Design Notes

[Link to Figma design in Dev mode](https://www.figma.com/design/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?node-id=4608-34986&m=dev&t=Pi7YPKvpXN1EUnDp-1)
[Link to Figma prototype](https://www.figma.com/proto/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?page-id=4337%3A31742&node-id=4608-34986&viewport=1026%2C1336%2C0.13&t=WQf31EZElCHZrbzc-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=4608%3A34986)

_Screenshots below should not be used for working on this ticket - always refer to Figma design for most up-to-date designs:_

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- Refactor "About the Condition" section to be Condition Summary. All Reportable conditions are now accordion in that section 

## Related Issue
Fixes #2184 

## Additional Information
General screenshots:
image
image
image
image

* Scenario 1:
  * eCR with just 1 reportable condition in the Viewer
  * http://localhost:3000/view-data?id=1.2.840.114350.1.13.297.3.7.8.688883.567479

https://github.com/user-attachments/assets/bee4afb9-98ad-4157-af7e-4636a5f42756

* Scenario 2:
  * eCR with n>1 reportable conditions in the Viewer AND pass in a SNOMED condition code that corresponds to one of the reportable conditions
  * http://localhost:3000/view-data?id=1.2.840.114350.1.13.478.3.7.8.688883.230886&snomed-code=840539006

https://github.com/user-attachments/assets/dc43511c-0fba-4f32-aa3b-b9c34fa9127d

* Scenario 3: 
  * eCR with n>1 reportable conditions in the Viewer without passing in a condition code
  * http://localhost:3000/view-data?id=1.2.840.114350.1.13.478.3.7.8.688883.230886

https://github.com/user-attachments/assets/c9f40603-5b4b-4267-b162-622b7bc79104

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Comment by sarahtress** [comment]
@BobanL this looks great! only one small nit - could you increase the spacing between each accordion to more closely match the designs? (the pink area in this attached screenshot)
Screenshot 2024-07-25 at 10 03 04 AM

**Comment by BobanL** [comment]
> @BobanL this looks great! only one small nit - could you increase the spacing between each accordion to more closely match the designs? (the pink area in this attached screenshot) Screenshot 2024-07-25 at 10 03 04 AM

Fixed! 
image

File: containers/ecr-viewer/src/app/services/ecrSummaryService.tsx [hunk 1]
```diff
@@ -105,80 +107,100 @@ export const evaluateEcrSummaryEncounterDetails = (
   ];
 };
 
+const evaluateRuleSummaries = (observation: Observation): Set<string> => {
```
Comment (angelathe) [inline containers/ecr-viewer/src/app/services/ecrSummaryService.tsx:110]: Could you add a JSDoc for this?

**Review by angelathe — APPROVED** [review]
Nice!! Just comment about adding a JSDoc

**Pull request merged**